app/shop.py

```diff
@@ -0,0 +1,34 @@
+import datetime
+
+
+class Shop:
+    def __init__(
+            self,
+            name: str,
+            location: list[int],
+            products: dict
+    ) -> None:
+        self.name = name
+        self.location = location
+        self.products = products
+        self.each_product_cost = {}
+        self.total_cost = 0
+
+    def purchase(self, product_cart: dict) -> int:
+        for product in product_cart:
+            product_cost = product_cart[product] * self.products[product]

```

Comment:
There is a potential issue here: if a product in `product_cart` is not available in `self.products`, this line will raise a KeyError. Consider adding a check to ensure the product exists in `self.products` before attempting to access its price.